### ISSUE TITLE
Implement a Wing Daemon to handle resource cleanup

### ISSUE THREAD

**Issue by skyrpex** [winglang/wing#4266]
### Feature Spec

The Wing Simulator will create resources such as HTTP servers or Docker containers, and in some cases it may be unable to clean them up (ie, the simulator process is unexpectedly killed).

The Wing CLI now implements a daemon process that will take care of cleaning resources in the background. Such resources include:

- Temporary directories that must be emptied or deleted
- Docker containers that must be stopped and/or deleted
- Dangling processes that run in the background and must be stopped

### Use Cases

It's very common to have dozens of Docker containers lying around after using the Simulator, and they are never deleted. Same thing may happen with other kind of resources.

With a daemon, these resources can be cleaned up easily.

### Implementation Notes

The daemon is a process that will be launched in the background while the simulator is starting.

The daemon will expose an HTTP interface that the simulator will use to report the resources that are constructed and need cleanup after the simulator reloads or stops.

The daemon will clean everything and stop itself if the simulator doesn't ping it after a certain amount of time.

### Component

CLI

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by MarkMcCulloh** [comment]
I'm not entirely against a daemon, but I think it can too easily cover up mistakes that should be resolved where they came from. We still want the simulator to be responsible for external resources, right? This is why things like SIGTERM and SIGINT exist. Receiving SIGKILL should come with an expectation that things won't be cleaned up (this even applies to child processes AFAIK).

I think a better first step would be to centralize how external resources are used and handle their cleanup by via OS signals. Worst case, we should inform the user if we weren't able to clean things up.

**Comment by eladb** [comment]
As I see it, I think cloud.Service essentially has a daemon api. OnStop should take care of cleanups. Isn't this sufficient?

**Comment by MarkMcCulloh** [comment]
Yeah that's the proposal I mentioned above rather than a daemon.
Notably we do not properly handle process kill signals right now, which is what we'd need to fix in the sim.

**Comment by eladb** [comment]
> Notably we do not properly handle process kill signals right now...

Definitely! Is there an issue for this? I think that's the first priority.
